pkg/proxy/register_test.go

```diff
@@ -117,3 +117,76 @@ func createRegister(r router.Router) *Register {
 
 	return register
 }
+
+func TestParametersReplacement(t *testing.T) {
+	log.SetOutput(ioutil.Discard)
+
+	def := &Definition{
+		ListenPath: "/api/recipes/{id:[\\da-f]{24}}",
+		Upstreams: &Upstreams{
+			Balancing: "roundrobin",
+			Targets:   []*Target{{Target: "http://localhost:9089/recipes/{id}"}},
+		},
+		Methods:    []string{"ALL"},
+		StripPath:  false,
+		AppendPath: false,
+	}
+	route := NewRoute(def)
+
+	chiRouter := router.NewChiRouter()
+
+	statsClient, _ := stats.NewClient("noop://", "")
+	register := NewRegister(chiRouter, Params{StatsClient: statsClient})
+	register.Add(route)
+
+	ts := test.NewServer(chiRouter)
+	defer ts.Close()
+
+	res, err := ts.Do("GET", "/api/recipes/5252b1b5301bbf46038b473f", make(map[string]string))
+	if err != nil {
+		t.Fatal("Call failed", err.Error())
+	}
+
+	defer res.Body.Close()
+
+	assert.Equal(t, 200, res.StatusCode)
+
+	body, _ := ioutil.ReadAll(res.Body)
+	assert.Equal(t, "{\"slug\":\"I'm a slug\"}", string(body))
+}
+
+func TestParametersReplacementWhenNoParameterIsAvailable(t *testing.T) {
+	log.SetOutput(ioutil.Discard)
+
+	def := &Definition{
+		ListenPath: "/api/recipes/search",
+		Upstreams: &Upstreams{
+			Balancing: "roundrobin",
+			Targets:   []*Target{{Target: "http://localhost:9089/recipes/{id}"}},
+		},
+		Methods:    []string{"ALL"},
+		StripPath:  false,
+		AppendPath: false,
+	}
+	route := NewRoute(def)
+
+	chiRouter := router.NewChiRouter()
+
+	statsClient, _ := stats.NewClient("noop://", "")
```

Comment:
Same regarding errors

Reply:
no need to check for error in tests for `noop` backend